### PR TITLE
[http] Improve escaping of URLs

### PR DESCRIPTION
**Pull request by J-N-K**
Related to #12350 
Related to #17039 

@jamesmelville FYI

### CODE REVIEW

File: bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/Util.java
Comment (lsiepel) [inline bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/Util.java:67]: `queryString` does not seem to be used.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Otherwise LGTM. Will also wait for @jamesmelville to comment before merge.

**Review by lsiepel — APPROVED** [review]

**Review by jamesmelville — APPROVED** [review]
Sorry I'm only on a phone but I think this looks good. I think you've now made the change backwards compatible for users using the old escapedURL flag which is great. Minor comments on naming. Thanks @J-N-K @lsiepel

**Comment by J-N-K** [comment]
@jamesmelville Thanks!

**Pull request merged**
